exporter/prometheusremotewriteexporter/wal.go

```diff
@@ -236,6 +246,28 @@ func (prweWAL *prweWAL) run(ctx context.Context) (err error) {
 	return nil
 }
 
+func (prweWAL *prweWAL) recordLagLoop(ctx context.Context, logger *zap.Logger) {
+	ticker := time.NewTicker(15 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-prweWAL.stopChan:
+			return
+		case <-ticker.C:
+			logger.Info("recording WAL lag")
+			prweWAL.mu.Lock()
+			wIndex := prweWAL.wWALIndex.Load()
+			rIndex := prweWAL.rWALIndex.Load()
```

Comment:
This is complete "shot in the dark", but when I see the methods `Load()` and `Store()` to read or write a number, I assume the underlying algorithm is using [Compare And Swap (CAS)](https://en.wikipedia.org/wiki/Compare-and-swap). CAS uses atomic operations, which means [we might not need to use Mutexes at all here](https://en.wikipedia.org/wiki/Compare-and-swap#:~:text=This%20operation%20is%20used%20to%20implement%20synchronization%20primitives%20like%20semaphores%20and%20mutexes%2C%5B1%5D%20as%20well%20as%20more%20sophisticated%20lock%2Dfree%20and%20wait%2Dfree%20algorithms.) 🤔

Reply:
I need to dig a bit deeper in the code to understand if that's really true. If it is, we can remove the mutexes in a follow up PR :)